pkg/controllerhelpers/handlers.go

```diff
@@ -202,3 +203,36 @@ func (h *Handlers[T]) HandleDeleteWithDepth(depth int, obj any, enqueueFunc Enqu
 func (h *Handlers[T]) HandleDelete(obj any, enqueueFunc EnqueueFuncType) {
 	h.HandleDeleteWithDepth(2, obj, enqueueFunc)
 }
+
+// DeletedStronglyTypedFinalStateUnknown is like DeletedFinalStateUnknown, but
+// uses a types.NamespacedName as the key. This is useful for objects that are
+// namespaced, as it allows for more precise identification of the object.
+type DeletedStronglyTypedFinalStateUnknown struct {
+	Key types.NamespacedName
+	Obj interface{}
+}
```

Comment:
```suggestion
type DeletedFinalStateUnknown struct {
	Key types.NamespacedName
	Obj interface{}
}
```
it's not any more "strongly typed" than the regular `cache.DeletedFinalStateUnknown`, so I'd suggest just redefining it here, the name collision is fine as we'll just use our implementation

Reply:
Renamed